core/src/test/java/org/mule/processor/SedaStageInterceptingMessageProcessorTestCase.java

```diff
@@ -191,6 +192,63 @@ public MuleEvent answer(InvocationOnMock invocation) throws Throwable
     }
 
     @Test
+    public void testProcessOneWayNegativeThreadWaitTimeout() throws Exception
+    {
+        final int threadTimeout = -1;
+        ThreadingProfile threadingProfile = new ChainedThreadingProfile(muleContext.getDefaultThreadingProfile());
+        threadingProfile.setThreadWaitTimeout(threadTimeout);
+        // Need 2 threads: 1 for polling, 1 to process work
+        threadingProfile.setMaxThreadsActive(2);
+        threadingProfile.setPoolExhaustedAction(WHEN_EXHAUSTED_WAIT);
+        threadingProfile.setMuleContext(muleContext);
+
+        queueProfile.setMaxOutstandingMessages(1);
+
+        final Latch latch = new Latch();
+        MessageProcessor mockListener = mock(MessageProcessor.class);
+        when(mockListener.process((MuleEvent)any())).thenAnswer(new Answer<MuleEvent>()
+        {
+            public MuleEvent answer(InvocationOnMock invocation) throws Throwable
+            {
+                // Await on latch so that only thread available is busy and other events have been queued.
+                latch.await();
+                return (MuleEvent)invocation.getArguments()[0];
+            }
+        });
+
+        SedaStageInterceptingMessageProcessor sedaStageInterceptingMessageProcessor = new SedaStageInterceptingMessageProcessor(
+            "testProcessOneWayNegativeThreadWaitTimeout", "testProcessOneWayNegativeThreadWaitTimeout", queueProfile,
+            queueTimeout, threadingProfile, queueStatistics, muleContext);
+        sedaStageInterceptingMessageProcessor.setListener(mockListener);
+        sedaStageInterceptingMessageProcessor.initialise();
+        sedaStageInterceptingMessageProcessor.start();
+
+        MessagingExceptionHandler exceptionHandler = mock(MessagingExceptionHandler.class);
+        Flow flow = mock(Flow.class);
+        when(flow.getExceptionListener()).thenReturn(exceptionHandler);
+        when(flow.getProcessingStrategy()).thenReturn(new AsynchronousProcessingStrategy());
+        final MuleEvent event = getTestEvent(TEST_MESSAGE, flow, MessageExchangePattern.ONE_WAY);
+
+        int NUMBER_OF_EVENTS = 3;
+
+        for (int i = 0; i < NUMBER_OF_EVENTS; i++)
+        {
+            sedaStageInterceptingMessageProcessor.process(event);
+        }
+
+        // Release latch only after 400ms which is double the default queue timeout.  If the new custom queueTimeout wasn't being
+        // used then event 3 would be rejected, as there is no room for it in the queue while event 1 processing is waiting on
+        // the latch.
+        Thread.sleep(400);
```

Comment:
Instead of waiting for x time, you can sometimes depend on another event that can be used to detect that the previous one has occurred. If that is not the case, then I would suggest you to use a timeout =1Ms instead